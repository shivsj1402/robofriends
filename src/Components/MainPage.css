@import url('https://fonts.googleapis.com/css2?family=Carter+One&display=swap');

h1{
    font-family: 'Carter One', cursive;
    font-size:60px;
    font-weight:200;
    color:#183e80;
}