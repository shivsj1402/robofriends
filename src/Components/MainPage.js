import React from 'react';
import CardList from './cardList'
import SearchBox from './searchBox'
import Scroll from './scroll'
import ErrorBoundaries from "./errorBoudaries";
import 'tachyons';
import './MainPage.css';
import Header from './Header';

class MainPage extends React.Component {

  componentDidMount(){
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    const { robots, serachField } = this.props;
    return robots.filter(robot => {
      return robot.name.toLowerCase().includes(serachField.toLowerCase());
    })
  }

  render(){ 
    const { onSearchChange, isPending} = this.props;
    if(isPending){
      return <h1>LOADING</h1>
    }else{
      return (
        <div className='tc'>
          <Header />
          <SearchBox searchChange={onSearchChange}/>
          <Scroll>
            <ErrorBoundaries>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundaries>
          </Scroll>
        </div>
      );
    }
  }
}

export default MainPage;
