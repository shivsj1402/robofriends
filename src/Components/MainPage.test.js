import React from "react";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainPage from "./MainPage";

let wrapper;
beforeEach(()=>{
    const mockProps ={
        onRequestRobots : jest.fn(),
        serachField : '',
        onSearchChange : jest.fn(), 
        robots: [], 
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps}/>);
})

it('Mainpage component render', ()=>{
    expect(toJson(wrapper)).toMatchSnapshot();
})

it('filters robots correctly', ()=>{
    const mockProps2 ={
        onRequestRobots : jest.fn(),
        serachField : 'a',
        onSearchChange : jest.fn(), 
        robots: [{
            id:3,
            name:'john',
            email: 'john@gmail.co,',
        }], 
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2}/>);
    expect(wrapper.instance().filterRobots()).toEqual([]);
    expect(wrapper2.instance().filterRobots()).toEqual([]);
})