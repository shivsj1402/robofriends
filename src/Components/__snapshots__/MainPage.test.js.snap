// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mainpage component render 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox
    searchChange={[MockFunction]}
  />
  <Scroll>
    <ErrorBoundaries>
      <CardList
        robots={Array []}
      />
    </ErrorBoundaries>
  </Scroll>
</div>
`;
